### PR TITLE
first hometask (menu and theme hooks)

### PR DESCRIPTION
**Pull request by dakruchko**

### CODE REVIEW

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    'access protected content' => [
+      'title' => t('lesson_menu module - access protected content'),
+      'description' => t('permission for page with protected content'),
+    ],
+  ];
+
+  return $permissions;
+}
+
+/**
```
Comment (dprudnikov) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Refactor

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+function two_load($val) {
+  $two = $val; // executing some heavy operations and returning prepared argument
+
+  return $two;
+}
+
+/**
+ * Load wildcard handler.
+ */
+function argument_load($val) {
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Module's name?

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+  return $val;
+}
+
+/**
+ * Implements hook_theme().
+ */
+function lesson_menu_theme() {
+  $theme['lesson_menu_render_images'] = [
+    'variables' => ['list' => NULL, 'generated_images' => NULL, 'original_image' => NULL],
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: split into multiline array

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+function lesson_menu_render_tabs($index = '') {
+  return ($index === '') ? t('This is the "tabs" menu entry') : t('This is the tab "@index" in the "basic tabs" example', ['@index' => $index]);
+}
+
+/**
+ * Page callback: Displays a page with passed arguments.
+ *
+ * @see lesson_menu_menu().
+ */
+function lesson_menu_render_arguments($one = '', $two = '') {
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: Refactor

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+  ]);
+}
+
+/**
+ * Page callback: Displays a page with content that depends on user's permission.
+ *
+ * @see lesson_menu_menu().
+ */
+function lesson_menu_render_access() {
+  return (user_access('access protected content')) ? t('Custom access for rendered content') : t('Custom access inversed for rendered content');
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: Remove unnecessary curves

**Review by dwebpoint — APPROVED** [review]

**Pull request merged**